packages/lisk-transactions/src/transactions/3_vote_transaction.ts

```diff
@@ -0,0 +1,483 @@
+		super(tx);
+		const typeValid = validator.validate(voteAssetTypeSchema, tx.asset);
+		const errors = validator.errors
+			? validator.errors.map(
+					error =>
+						new TransactionError(
+							`'${error.dataPath}' ${error.message}`,
+							tx.id,
+							error.dataPath,
+						),
+			  )
+			: [];
+		if (!typeValid) {
+			throw new TransactionMultiError('Invalid field types', tx.id, errors);
+		}
+		this.asset = tx.asset as VoteAsset;
+	}
+
+	public static create(input: CastVoteInput): object {
+		validateInputs(input);
+		const { passphrase, secondPassphrase, votes = [], unvotes = [] } = input;
+
+		const plusPrependedVotes = prependPlusToPublicKeys(votes);
+		const minusPrependedUnvotes = prependMinusToPublicKeys(unvotes);
+		const allVotes: ReadonlyArray<string> = [
+			...plusPrependedVotes,
+			...minusPrependedUnvotes,
+		];
+
+		const transaction = {
+			...createBaseTransaction(input),
+			type: 3,
+			fee: VOTE_FEE.toString(),
+			asset: {
+				votes: allVotes,
+			},
+		};
+
+		if (!passphrase) {
+			return transaction;
+		}
+
+		const {
+			address: senderId,
+			publicKey: senderPublicKey,
+		} = getAddressAndPublicKeyFromPassphrase(passphrase);
+		const transactionWithSenderInfo = {
+			...transaction,
+			senderId,
+			senderPublicKey,
+			recipientId: senderId,
+		};
+
+		const voteTransaction = new VoteTransaction(transactionWithSenderInfo);
+		voteTransaction.sign(passphrase, secondPassphrase);
+
+		return voteTransaction.toJSON();
+	}
+
+	public static fromJSON(tx: TransactionJSON): VoteTransaction {
+		const transaction = new VoteTransaction(tx);
+		const { errors, status } = transaction.validateSchema();
+
+		if (status === Status.FAIL && errors.length !== 0) {
+			throw new TransactionMultiError(
+				'Failed to validate schema.',
+				tx.id,
+				errors,
+			);
+		}
+
+		return transaction;
+	}
+
+	protected getAssetBytes(): Buffer {
+		return Buffer.from(this.asset.votes.join(''), 'utf8');
+	}
+
+	public assetToJSON(): object {
+		return {
+			votes: this.asset.votes,
+		};
+	}
+
+	public getRequiredAttributes(): Attributes {
+		const attr = super.getRequiredAttributes();
+		const publicKey = this.asset.votes.map(pkWithAction =>
+			pkWithAction.slice(1),
+		);
+
+		return {
+			[ENTITY_ACCOUNT]: {
+				...attr[ENTITY_ACCOUNT],
+				publicKey,
+			},
+		};
+	}
+
+	public verifyAgainstOtherTransactions(
+		transactions: ReadonlyArray<TransactionJSON>,
+	): TransactionResponse {
+		const sameTypeTransactions = transactions
+			.filter(
+				tx =>
+					tx.senderPublicKey === this.senderPublicKey && tx.type === this.type,
+			)
+			.map(tx => new VoteTransaction(tx));
+		const publicKeys = this.asset.votes.map(vote => vote.substring(1));
+
+		const errors = sameTypeTransactions.reduce(
+			(previous, tx) => {
+				if (
+					tx.asset.votes
+						.map(vote => vote.substring(1))
+						.some(publicKey => publicKeys.includes(publicKey))
+				) {
+					return [
+						...previous,
+						new TransactionError(
```

Comment:
Here, we should at least specify which vote is conflicting and which transaction is the current transaction conflicting with. In the current implementation, we will get multiple error messages with the same description without any useful information. What do you think?